src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -103,21 +110,20 @@ public void doGet(
    * @param nDays number of days unread emails should be checked for
    * @param mHours unread emails from the last mHours hours should be used to calculate the
    *     "unreadEmailsFromMHours" statistic
-   * @param gmailClient GmailClient implementation with valid credential
+   * @param gmailClient GmailClient implementation
    * @return GmailResponse object
    * @throws IOException if an issue occurs with the Gmail Service while obtaining emails
    */
   private GmailResponse generateGmailResponse(int nDays, int mHours, GmailClient gmailClient)
       throws IOException {
     GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.METADATA;
 
-    List<Message> unreadMessages =
-        GmailResponseUtility.getUnreadEmailsFromNDays(gmailClient, messageFormat, nDays);
+    List<Message> unreadMessages = gmailClient.getUnreadEmailsFromNDays(messageFormat, nDays);
     int unreadEmailsFromNDays = unreadMessages.size();
```

Comment:
Move the unread message count calculation into `GmailResponseHelper`. Otherwise, the separation of concerns looks really arbitrary. This is especially apparent when looking at the unit tests. The tests for unread e-mail count are subtly different from the tests for the other parts of the response, and it's confusing.

Reply:
RESOLVED